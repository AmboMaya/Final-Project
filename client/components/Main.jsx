import React from 'react'
import Statistics from './Statistics'
import CardList from './CardList'
import RegisterForm from './RegisterForm'
import LoginForm from './LoginForm'
import MyCalendar from './MyCalendar'
import {Route, BrowserRouter as Router, Switch} from 'react-router-dom'

export default class Main extends React.Component {
  render () {
    return (
      <Router>
        <Switch>
          <Route exact path='/' component={CardList} />
          <Route path='/photo' component={MyCalendar} />
          <Route path='/statistics' component={Statistics} />
          <Route path='/login' component={LoginForm} />
          <Route path='/auth' component={RegisterForm} />
        </Switch>
      </Router>
    )
  }
}
