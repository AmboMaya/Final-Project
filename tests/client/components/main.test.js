import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Main from '../../../client/components/Main'

configure({adapter: new Adapter()})

test('<Main />', () => {
  const expected = '<BrowserRouter />'
  const wrapper = shallow(<Main />)
  expect(wrapper.text()).toMatch(expected)
})

test('<Main /> matches the last snapshot', () => {
  const renderer = new ShallowRenderer()
  const actual = renderer.render(<Main />)
  expect(actual).toMatchSnapshot()
})
