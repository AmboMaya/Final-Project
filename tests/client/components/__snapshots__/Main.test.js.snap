/ Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> matches the last snapshot 1`] = `
<BrowserRouter>
  <Route
     component={[Function]}
     path=“/”
   />
    <Switch>
       <Route
         component={[Function]}
         exact={true}
         path=“/calendar”
       />
       <Route
         component={[Function]}
         path=“/statistics”
       />
       <Route
         component={[Function]}
         path=“/login”
       />
     </Switch>
 </BrowserRouter>